File: xamarin/android/PropertyCross/Utils/AsyncDrawable.cs
using System;
using Android.Graphics.Drawables;
using Android.Content.Res;
using Android.Graphics;

namespace com.propertycross.xamarin.android
{
	public class AsyncDrawable : BitmapDrawable
	{
		private WeakReference taskRef;

		public AsyncDrawable(Resources resources, Bitmap bitmap, DownloadImageTask task)
			: base(resources, bitmap)
		{
			taskRef = new WeakReference(task);
		}

		public DownloadImageTask GetTask()
		{
			return (DownloadImageTask) taskRef.Target;
		}
	}
}

File: native/android/PropertyCross/src/com/propertycross/android/util/AsyncDrawable.java
package com.propertycross.android.util;

import java.lang.ref.WeakReference;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;

public class AsyncDrawable extends BitmapDrawable {

	private WeakReference<DownloadImageTask> taskRef;
	
	public AsyncDrawable(Resources resources, Bitmap bitmap, DownloadImageTask task) {
		super(resources, bitmap);
		taskRef = new WeakReference<DownloadImageTask>(task);
	}
	
	public DownloadImageTask getTask() {
		return taskRef.get();
	}
	
}
